internal/graph/resolve_check_request.go

```diff
@@ -39,13 +48,63 @@ type ResolveCheckRequestMetadata struct {
 	WasThrottled *atomic.Bool
 }
 
+type ResolveCheckRequestParams struct {
```

Comment:
(No action item) I feel a bit iffy about having to structs (ResolveCheckRequest and ResolveCheckRequestParams) that are so similar. Maybe we can revisit this later